lib/dry/system/auto_registrar.rb

```diff
@@ -34,9 +35,15 @@ def finalize!
       # @api private
       def call(dir)
         registration_config = Configuration.new
+
         yield(registration_config) if block_given?
+
         components(dir).each do |component|
-          next if !component.auto_register? || registration_config.exclude.(component)
+          next if (
+            registered?(component) ||
```

Comment:
@timriley Should these inline conditions be extracted in one or more private methods to let the intent to emerge?

Reply:
@jodosha Fair call :) Addressed in https://github.com/dry-rb/dry-system/pull/155/commits/515f76e6c09e3f5c0bcd1622f3beb3517701575c